### PR TITLE
Implementing tabsv2

### PR DESCRIPTION
**Pull request by abottega**
Refactored Tabs component to have better responsive pattern and better config use

**Before:**
Screenshot 2024-09-26 at 2 00 43 PM

**After:**
Screenshot 2024-09-26 at 2 01 03 PM

### CODE REVIEW

**Review by JoshuaCarter — DISMISSED** [review]
I'm not that familiar with orcs so I may be off base with comments. Leave to your judgement. I kinda feel like the 'more' button could afford to be more visually destincy compared to the tabs, but that's just my opinion.

File: lib/components/Tabs/index.js [hunk 1]
```diff
@@ -1,145 +1,277 @@
...
+
+  const showMoreButtonRef = useRef();
+
+  const containerVisibleWidth = useRef(0);
+
+  const [showMoreTabs, setShowMoreTabs] = useState([]);
+
+  const calculateVisibility = useCallback(
+    (actionElements) => {
+      const showMoreButtonWidth = showMoreButtonRef.current
```
Comment (KitoC) [inline lib/components/Tabs/index.js:138]: This line can be `const showMoreButtonWidth = showMoreButtonRef?.current?.offsetWidth ?? 0`

File: lib/components/Tabs/index.js [hunk 2]
```diff
@@ -1,145 +1,277 @@
...
+        ? showMoreButtonRef.current.offsetWidth
+        : 0;
+      // as we loop through the tabs, we need to calculate the width of the visible tabs.
+      let calculatedWidth = showMoreTabs.length ? showMoreButtonWidth : 0;
+
+      // variable for the list of hidden tabs which will be put to react state
+      const newShowMoreTabs = [];
+
+      [...actionElements]
+        .filter((el) => el.tagName === "A")
```
Comment (KitoC) [inline lib/components/Tabs/index.js:148]: Add a comment for this filter "this ensures that the ShowMore button is not included in this logic"

**Review by KitoC — DISMISSED** [review]
Looks good, a few small rec's

File: lib/components/Tabs/index.js [hunk 3]
```diff
@@ -1,145 +1,277 @@
...
-        notification ? "notification notification-" + notification : ""
-      }`}
-      {...props}
-    >
-      {React.Children.map(children, (child) =>
-        React.cloneElement(child, { role: "tab", "aria-selected": `${active}` })
-      )}
-    </TabItem>
+    },
+    [showMoreTabs]
```
[the remark this answers is not in the document]
Comment (KitoC) [reply lib/components/Tabs/index.js]: Oh wait, refs in dep arrays are an antipattern and also the method returned from `useState` never changes either (it is a dispatch). Eslint does not complain about these not being present.

**Review by KitoC — APPROVED** [review]

File: lib/components/Tabs/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (JoshuaCarter) [reply lib/components/Tabs/index.js:162]: Yea good point (I haven't used refs in a while), and I didn't assess this closely (e.g. I didn't realise that `containerVisibleWidth` was a ref). Apparently state setters are guaranteed stable too (I don't think app linting reflects that?).

**Pull request merged**